core/silkworm/state/state.hpp

```diff
@@ -38,6 +36,14 @@ class State {
     /** @name Readers */
     ///@{
 
+    // methods from BlockState
```

Comment:
There's no need to repeat `read_header` & `read_body` methods here, I'd remove their declarations from `State` altogether.

Reply:
done